README.md

```diff
@@ -216,6 +217,45 @@ If you use the `HttpBodyStream` the whole transfer will be [chunked encoded](htt
 
 Check out the `examples` folder how your computation could look like.
 
+### Streaming requests
+
+Streaming requests makes it possible to send big amount of data in small chunks from the client to the server. E.g you can start the computation of the request,
+when your application received an specific part of the body.
+
+The body of the request object in your callback and middleware function will be a [ReadableStreamInterface](https://github.com/reactphp/stream).
+
+```php
+$callback = function (RequestInterface $request) {
+    $body = $request->getBody();
+    
+    return new Promise(function ($resolve, $reject) use ($body) {
+        $content = '';
+        $body->on('data', function ($chunk) use ($resolve, &$content) {
+            $content .= "hello";
```

Comment:
What does this do? :-)

Perhaps add some description and/or use the one from examples/ folder?